packages/core/src/thing-description.ts

```diff
@@ -0,0 +1,258 @@
+/********************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ * Document License (2015-05-13) which is available at
+ * https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+
+// global W3C WoT Scripting API definitions
+import * as WoT from "wot-typescript-definitions";
+import * as TDT from "wot-thing-description-types";
+
+export const DEFAULT_CONTEXT_V1 = "https://www.w3.org/2019/wot/td/v1";
+export const DEFAULT_CONTEXT_V11 = "https://www.w3.org/2022/wot/td/v1.1";
+export const DEFAULT_CONTEXT_LANGUAGE = "en";
+export const DEFAULT_THING_TYPE = "Thing";
+
+type DeepPartial<T> = T extends Record<string, unknown>
+    ? {
+          [P in keyof T]?: T[P] extends Array<infer I> ? Array<DeepPartial<I>> : DeepPartial<T[P]>;
+      }
+    : T;
+export type ThingModel = DeepPartial<Thing>;
```

Comment:
Spotted a possible collision with the concept of `ThingModel` from `@thingweb/thing-model`. Is it the same concept? or something else?

Reply:
Mhh, what you mean by that. I took it over from

https://github.com/eclipse-thingweb/node-wot/blob/274bf4aada923429e9164cfe8a82cd387ac6e6ea/packages/td-tools/src/td-tools.ts#L23-L28

Do you think this should not be needed?

Reply:
With https://github.com/eclipse-thingweb/node-wot/pull/1312/commits/8c1e0598f4c45cb42e60b292013db3cf2e31516a I removed our own definition of `ThingModel`. Instead I used the one from "wot-thing-model-types"

I think this should resolve your concerns...